scripts/pinned_build.sh

```diff
@@ -57,13 +57,42 @@ popdir() {
 }
 
 try(){
+    echo "$ $*"
     "$@"
     res=$?
     if [[ ${res} -ne 0 ]]; then
-        exit 255
+        exit $res
     fi
 }
 
+install_dependencies() {
+    echo 'Installing package dependencies.'
+    if [[ "$(uname)" == 'Linux' && -f /etc/debian_version ]]; then
+        if [[ "$(id -u)" != '0' ]]; then # if not root, use sudo for the package manager
+            SUDO_CMD='sudo'
```

Comment:
afaik sudo is not guaranteed to be installed

Reply:
> afaik sudo is not guaranteed to be installed

Truth. And not everyone likes `sudo`, either. It might be appropriate to test for `sudo` and report an error if it's missing.

Reply:
Done.